flink-core/src/test/java/org/apache/flink/testutils/assertj/FlinkAsserts.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.testutils.assertj;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * {@code FlinkAsserts} collect {@code assertThat} factory methods for any custom {@link
+ * org.assertj.core.api.AbstractAssert} implementation.
+ */
+public class FlinkAsserts {
+
+    public static CauseTreeContainsAssert assertThat(Throwable actualThrowable) {
+        return new CauseTreeContainsAssert(actualThrowable);
+    }
+
+    public static <T> FlinkCompletableFutureAssert<T> assertThat(CompletableFuture<T> actual) {
+        return new FlinkCompletableFutureAssert<>(actual);
+    }
```

Comment:
```suggestion
    private static final InstanceOfAssertFactory<Throwable, CauseTreeContainsAssert> CAUSE_TREE_FACTORY =
            new InstanceOfAssertFactory<>(
                    Throwable.class,
                    CauseTreeContainsAssert::new);

    public static InstanceOfAssertFactory<Throwable, CauseTreeContainsAssert> exceptionHierarchy() {
        return CAUSE_TREE_FACTORY;
    }
```

Reply:
Good point. I like that. I used a public static field in the implementing class, though, to reduce the boilerplate. WDYT?

Reply:
If we decide to add more "matchers" in the future, would that make them harder to discover? 

I'm thinking along this line: If we'd have all the factories accessible in the same class, would it make developer experience / discovery of the matchers you could leverage smoother?